test/LoggerServiceFactoryTest.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Log;
+
+use Closure;
+use Laminas\Db\Adapter\Adapter;
+use Laminas\Log\Logger;
+use Laminas\Log\LoggerServiceFactory;
+use Laminas\Log\Processor\ProcessorInterface;
+use Laminas\Log\ProcessorPluginManager;
+use Laminas\Log\Writer\Db as DbWriter;
+use Laminas\Log\Writer\MongoDB as MongoDBWriter;
+use Laminas\Log\Writer\Noop;
+use Laminas\Log\Writer\WriterInterface;
+use Laminas\Log\WriterPluginManager;
+use Laminas\ServiceManager\Config;
+use Laminas\ServiceManager\Exception\ServiceNotFoundException;
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use Laminas\ServiceManager\ServiceManager;
+use MongoDB\Driver\Manager;
+use PHPUnit\Framework\TestCase;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+use function count;
+use function extension_loaded;
+
+class LoggerServiceFactoryTest extends TestCase
```

Comment:
I think we need a few additional tests:

- One for when `log.writers` is not iterable.
- One for when `log.writers[]` are not arrays/`ArrayAccess`